### PR TITLE
Implement stupid computer player - swift code only

### PR DESCRIPTION
**Pull request by merecarvill**

### CODE REVIEW

**Comment by ericmeyer** [comment]
Make sure to be mindful of keeping a clean Git history.

I see a "wip" commit, as well as removal of some spike code. If you want to save spike code for later, I would recommend creating and pushing a new branch. It is best to keep the master branch clean.

**Comment by ericmeyer** [comment]
Also, the addition of the comments should probably have gone into a separate branch as well.

Although probably overkill for this project, one could create GitHub issues and/or stories for some of the feedback as appropriate. This is more useful on a larger / longer running project.

File: TicTacToeApp/ComputerPlayer.swift [hunk 1]
```diff
@@ -0,0 +1,25 @@
+
+import Foundation
+
+public class ComputerPlayer {
+
+  public init() {
```
Comment (ericmeyer) [inline TicTacToeApp/ComputerPlayer.swift:6]: IMO, an empty `public init` that is required by the compiler might be one of the rare times a one-liner method is OK. As long as you are consistent, either way is fine with me. A quick look in a couple style-guides didn't show anything in either direction.

``` swift
public init() {}
```

File: TicTacToeApp/ViewController.swift [hunk 2]
```diff
@@ -67,15 +105,27 @@ public class ViewController: UIViewController {
...
-  private func updateGameBoardButtonForPlayerMove(button: UIButton, playerMark: PlayerMark) {
+  private func updateGameBoardForPlayerMove(spaceId: Int, playerMark: PlayerMark) {
+    let button = getButtonByTag(spaceId) ?? UIButton()
+
     button.setTitle(getButtonTextFor(playerMark), forState: UIControlState.Normal)
     button.enabled = false
   }
 
+  private func getButtonByTag(spaceId: Int) -> UIButton? {
+    for button in boardButtons {
```
Comment (ericmeyer) [inline TicTacToeApp/ViewController.swift:120]: There might be a more Swift-y way to do this.

File: TicTacToeApp/ViewController.swift [hunk 3]
```diff
@@ -91,4 +141,12 @@ public class ViewController: UIViewController {
   private func getButtonTextFor(playerMark: PlayerMark) -> String {
     return playerMark == PlayerMark.X ? "X" : "O"
   }
+
+  private func getModeFrom(button: UIButton) -> GameMode {
+    switch button.tag {
+      case 0: return GameMode.HumanVsHuman
```
Comment (ericmeyer) [inline TicTacToeApp/ViewController.swift:147]: Do tags have to be integers?

File: TicTacToeApp/ComputerPlayer.swift (hunk 1 above)
Comment (merecarvill) [inline TicTacToeApp/ComputerPlayer.swift:6]: OK, good to know.

File: TicTacToeApp/ViewController.swift (hunk 2 above)
Comment (merecarvill) [inline TicTacToeApp/ViewController.swift:120]: Yeah, it doesn't strike me as particularly elegant, but this one-liner works:

return boardButtons.filter{ $0.tag == spaceId }.first

File: TicTacToeApp/ViewController.swift (hunk 3 above)
Comment (merecarvill) [inline TicTacToeApp/ViewController.swift:147]: Yep.

**Pull request merged**